autokeras/hypermodel/hyper_block.py

```diff
@@ -41,9 +41,20 @@ def build(self, hp, inputs=None):
         output_node = Flatten().build(hp, output_node)
 
         for i in range(hp.Choice('num_layers', [1, 2, 3], default=2)):
-            output_node = tf.keras.layers.Dense(hp.Choice('units_{i}'.format(i=i),
-                                                          [16, 32, 64],
-                                                          default=32))(output_node)
+            if hp.Choice('layer_category', ['Dense','BN','Dropout','Activation'], default='Dense') == 'Dense':
```

Comment:
As long as you use the same name of 'layer_category', the value of it will always be the same. So it would be all dense, or all dropout, or all activation.
I think we can do it like the following:

`for ... 
   layer_stack = hp.Choice('layer_stack', ['dense-bn-act', 'dense-act', 'act-bn-dense']
   if layer_stack == ...:
   elif layer_stack == ...:

Reply:
If we use this kind of architecture, it means the search space is much less than before. Because we are just stacking these 'daense-bn-act'. So  I think we just give them a different name is OK. For example, 'layer_category_Dense','layer_category_BN'. So this will totally make a different. Each loop, we will get a different layer.

Reply:
You are right that the search space is restricted. However, that is what we want for now. Currently, the search algorithms are not efficient enough. So we want to restrict the space to a small one that most of the architectures in it are good.